### ISSUE TITLE
Implement `StoreErrorNotice` component

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#1749]
## Feature Description

Almost all of our Site Kit modules currently implement an `ErrorNotice` component, which works the same way across and thus is duplicated throughout them. We should implement a single `StoreErrorNotice` component instead and remove all the duplicates.

While the `getError` selector is not technically abstracted yet, let's keep that as is for now. Having this component is already a small win for DRY.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* There should be a `StoreErrorNotice` JS component in `assets/js/components/StoreErrorNotice.js` which should use almost the same code as the existing `ErrorNotice` components (e.g. `assets/js/modules/tagmanager/components/common/ErrorNotice.js`).
* The only difference from the existing components is that it should receive the store name as `storeName` prop.
* All modules should be adjusted so that their `ErrorNotice` component is removed and they instead use the new `StoreErrorNotice` (e.g. `<StoreErrorNotice storeName="modules/analytics" />`).

## Implementation Brief

* Create `assets/js/components/StoreErrorNotice.js` from `assets/js/modules/analytics/components/common/ErrorNotice.js` and update imports as a starting point
* Replace `STORE_NAME` with `storeName` from props
* Use `PropTypes` to mark `storeName` as a required string prop
* Accept an additional `shouldDisplayError` function as an optional prop
	- Should accept the `error` as the first and only parameter, and return a boolean (`true` to display, `false` not to)
	- defaults to `() => true`
	- Update existing conditional which returns early if there is no error or it is a permission scope error, to include `|| ! shouldDisplayError( error )`
* Replace all `ErrorNotice` usage in module components with `<StoreErrorNotice storeName={ STORE_NAME } />`, except AdSense (since there are so many uses of it)
	* For the AdSense `ErrorNotice`, update the component to use `StoreErrorNotice` internally and pass a function to `shouldDisplayError`  
	Basically `shouldDisplayError = useCallback( ( error ) => undefined === errorToStatus( error ), [] )`
* Delete all `ErrorNotice.js` components, except for the one for AdSense

## QA Brief

* Trigger errors in both module setup and settings for AdSense, Analytics, Tag Manager, and Optimize 
    - The easiest way to trigger generic errors is probably to disconnect from the internet or use [Request Blocking](https://developers.google.com/web/updates/2017/04/devtools-release-notes#block-requests) in Chrome devtools to block the `settings` datapoint.
* Ensure errors display as usual

## Changelog entry

* N/A

**Comment by felixarntz** [comment]
IB ✅

**Comment by eugene-manuilov** [comment]
QA :heavy_check_mark: The new component is added, old ErrorNotice components are removed, errors are displayed correctly.
